### DISCUSSION TITLE
Add typescript-plugin-css-modules to create-next-app --ts

### DISCUSSION THREAD

**Discussion by NoamNol** [vercel/next.js#33260] [Ideas]
### Describe the feature you'd like to request

[typescript-plugin-css-modules](https://github.com/mrmckeb/typescript-plugin-css-modules) should be included in `npx create-next-app@latest --ts` for better IDE experience with CSS Modules.
See: https://github.com/mrmckeb/typescript-plugin-css-modules/issues/34#issuecomment-1010264907

### Describe the solution you'd like

Include the typescript-plugin-css-modules.
Install it in `node_modules` and add it to `tsconfig.json` as described in the [README](https://github.com/mrmckeb/typescript-plugin-css-modules#installation).

### Describe alternatives you've considered

Add a clear explanation about it to the docs.

**Comment by balazsorban44** [comment]
You can add it in a PR as described here:

https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples

**Comment by NoamNol** [comment]
@balazsorban44 I think it should be part of `create-next-app --ts`.

**Comment by balazsorban44** [comment]
The `--ts` flag is used to simply install the default template but with TypeScript. Next.js handles CSS modules out-of-the-box in both cases (JS and TS). The mentioned plugin isn't necessary for Next.js.

It would still be acceptable to add a standalone example if someone needs it, I guess.

**Comment by NoamNol** [comment]
If Next.js handles CSS modules out-of-the-box with TypeScript, It makes sense to add the TypeScript safety to the CSS modules.

With the plugin you know when you type a wrong class name, and you can also use 'Go to definition'.
(Now 'Go to definition' works with JS but not with TS)

**Comment by felixcatto** [comment]
Also there is a webpack loader exist - [css-modules-typescript-loader](https://github.com/seek-oss/css-modules-typescript-loader)
It generates simply d.ts files and usually follow after css loader

```
  { loader: 'css-modules-typescript-loader' },
  { loader: 'css-loader', options: {...} },
```
